libs/mutantschemata/cpp_source/rewrite.hpp

```diff
@@ -0,0 +1,794 @@
+                } else {
+                    llvm::errs() << "new expression doesn't make sense! mutant not added\n";
+                }
+            }
+        }
+        
+        // check if we need to insert some singletons
+        for (const auto &elem: singletons) {
+            switch (elem) {
+                case Singleton::LHS:
+                    newExpr = lhs;
+                    newExpr += ": ";
+                    break;
+                case Singleton::RHS:
+                    newExpr = rhs;
+                    newExpr += ": ";
+                    break;
+                case Singleton::True:
+                    newExpr = "true: ";
+                    break;
+                case Singleton::False:
+                    newExpr = "false: ";
+                    break;
+                default:
+                    // continue for loop
+                    continue;
+            }
+            // insert the singleton
+            createAndStoreActualMutant(binOp, newExpr, endBracket);
+        }
+        
+    }
+    
+    void createAndStoreActualMutant(const clang::BinaryOperator *binOp, const std::string &newExpr, const std::string &endBracket) {
+        
+        
+        
+        // calculate offset and FileEntry (we don't use FileID as this can change depending on the order of opening files etc.)
+        const clang::FileEntry *fE;
+        unsigned exprOffs, bracketsOffs;
+        calculateOffsetLoc(binOp->getLocStart(), fE, exprOffs);
+        calculateOffsetLoc(binOp->getLocEnd(), fE, bracketsOffs, true);
+        
+        // create and store the created (meta-) mutant
+        InsertedMutant *im = new InsertedMutant(fE, exprOffs, bracketsOffs);
+        MutantInsert *mi = new MutantInsert(im, newExpr, endBracket);
+        
+        insertedMutants.value_comp() = insertedMutants.key_comp();
+        std::set<MutantInsert*, mutantLocComp>::iterator it = insertedMutants.find(mi);
+        if (it == insertedMutants.end()) {
+            mutantInserts.push_back(mi);
+            insertedMutants.insert(mi);
+            llvm::errs() << "inserted mutant in FE: " << std::to_string((long)(mi->mutantLoc->fE)) << " @offset: [" << mi->mutantLoc->exprOffs << ", " << mi->mutantLoc->bracketsOffs << "] | " << mi->expr << "\n";
+            mutant_count++; //increase count as we are sure it isn't a duplicated vallid mutant
+        } else {
+            llvm::errs() << "duplicate mutant in FE: " << std::to_string((long)(mi->mutantLoc->fE)) << " @offset: [" << mi->mutantLoc->exprOffs << ", " << mi->mutantLoc->bracketsOffs << "] | " << mi->expr << "\n";
+        }
+    }
+    
+    /**
+     * Check if we want to mutate this file.
+     * (if it's a sourceFile and we haven't yet visited it)
+     */
+    bool isfileToMutate(clang::FullSourceLoc FullLocation) {
+        const clang::FileEntry *fE = SourceManager->getFileEntryForID(SourceManager->getFileID(FullLocation));
+        if (fE) {
```

Comment:
The indention of these ifblocks does not follow the rest of the file where the indention is 4 spaces